windows/lcow_sandbox.go

```diff
@@ -0,0 +1,521 @@
+			svm.Unlock()
+			log.G(ctx).Debugf("%s (%s): releasing cachedScratchMutex on error path", title, context)
+			r.cachedScratchMutex.Unlock()
+
+			// Do a force terminate and remove it from the map on failure, ignoring any errors
+			if err2 := r.terminateServiceVM(ctx, id, "error path from CreateExt4Vhdx", true); err2 != nil {
+				log.G(ctx).Warnf("failed to terminate service VM on error path from CreateExt4Vhdx: %s", err2)
+			}
+
+			return nil, fmt.Errorf("failed to create SVM scratch VHDX (%s): %s", context, err)
+		}
+		log.G(ctx).Debugf("%s (%s): releasing serviceVM after %s created and cached to %s", title, context, scratchTargetFile, r.cachedScratchFile)
+		svm.Unlock()
+	}
+	log.G(ctx).Debugf("%s (%s): releasing cachedScratchMutex", title, context)
+	r.cachedScratchMutex.Unlock()
+
+	// Hot-add the scratch-space if not already attached
+	if !svm.scratchAttached {
+		log.G(ctx).Debugf("lcowdriver: startServiceVmIfNotRunning: (%s) hot-adding scratch %s - locking serviceVM", context, scratchTargetFile)
+		svm.Lock()
+		if err := svm.config.HotAddVhd(scratchTargetFile, toolsScratchPath, false, true); err != nil {
+			log.G(ctx).Debugf("%s (%s): releasing serviceVM on error path of HotAddVhd: %s", title, context, err)
+			svm.Unlock()
+
+			// Do a force terminate and remove it from the map on failure, ignoring any errors
+			if err2 := r.terminateServiceVM(ctx, id, "error path from HotAddVhd", true); err2 != nil {
+				log.G(ctx).Warnf("failed to terminate service VM on error path from HotAddVhd: %s", err2)
+			}
+
+			return nil, fmt.Errorf("failed to hot-add %s failed: %s", scratchTargetFile, err)
+		}
+		log.G(ctx).Debugf("%s (%s): releasing serviceVM", title, context)
+		svm.Unlock()
+	}
+
+	log.G(ctx).Debugf("lcowdriver: startServiceVmIfNotRunning: (%s) success", context)
+	return svm, nil
+}
+
+// getServiceVM returns the appropriate service utility VM instance, optionally
+// deleting it from the map (but not the global one)
+func (r *LCOWRuntime) getServiceVM(ctx context.Context, id string, deleteFromMap bool) (*serviceVMItem, error) {
+	log.G(ctx).Debugf("lcowdriver: getservicevm:locking serviceVmsMutex")
+	r.serviceVmsMutex.Lock()
+	defer func() {
+		log.G(ctx).Debugf("lcowdriver: getservicevm:releasing serviceVmsMutex")
+		r.serviceVmsMutex.Unlock()
+	}()
+	if r.globalMode {
+		id = svmGlobalID
+	}
+	if _, ok := r.serviceVms[id]; !ok {
+		return nil, fmt.Errorf("getservicevm for %s failed as not found", id)
+	}
+	svm := r.serviceVms[id]
+	if deleteFromMap && id != svmGlobalID {
+		log.G(ctx).Debugf("lcowdriver: getservicevm: removing %s from map", id)
+		delete(r.serviceVms, id)
+	}
+	return svm, nil
+}
+
+// terminateServiceVM terminates a service utility VM if its running, but does nothing
+// when in global mode as it's lifetime is limited to that of the daemon.
+func (r *LCOWRuntime) terminateServiceVM(ctx context.Context, id, context string, force bool) error {
```

Comment:
If the cached scratch vhd is created at plugin init as suggested, this can be simplified and moved to be a `serviceVMItem` method.

The only place that use it being `CreateReadWrite` the defer could just check that it's not in global mode before calling it.

It would simply be something like:
```go
func (svm *serviceVMItem) terminateServiceVM(ctx context.Context) error {
	// We run the deletion of the scratch as a deferred function to at
	// least attempt clean-up in case of errors.
	defer func() {
			if err := os.Remove(svm.scratchVhd); err != nil {
				log.G(ctx).WithError(err).WithFields(logrus.Fiedlds{"svm": svm.config.Name,
					"scratch-vhd": svm.scratchVhd,
				}).Warnf("failed to remove scratch file")
			}
	}()

	// Nothing to do if it's not running
	if svm.config.Uvm != nil {
		if err := svm.config.Uvm.Terminate(); err != nil {
			return errors.Wrapf(err, "failed to terminate utility VM %s", svm.config.Name)
		}

		timeout := time.Duration(svm.config.UvmTimeoutSeconds) * time.Second
		if err := svm.config.Uvm.WaitTimeout(timeout); err != nil {
			return errors.Wrapf("failed waiting for utility VM to terminate %s", svm.config.Name)
		}
	}

	return nil
}
```

This is assuming that `scratAttached bool` is replaced by `scratchVhd string` in the definition of`serviceVMItem`